kimchi/src/circuits/polynomials/generic.rs

```diff
@@ -330,9 +331,14 @@ pub mod testing {
                 let mul = self.coeffs[coeffs_offset + 3]
                     * this[register_offset]
                     * this[register_offset + 1];
+                let public = if coeffs_offset == 0 {
+                    public.get(row).cloned().unwrap_or(F::zero())
+                } else {
+                    F::zero()
+                };
                 ensure_eq!(
                     zero,
-                    sum + mul + self.coeffs[coeffs_offset + 4],
+                    sum + mul + self.coeffs[coeffs_offset + 4] - public,
```

Comment:
I don't think this is right: the public input is already accounted for as part of the witness, and the gates for the public input should already be at the beginning of the witness. The old version was already correct.

Reply:
I'm not sure I understand. If you don't remove the public element, which is in the witness, then the sum is not going to be equal to zero.

For the first `public` rows the equation is going to be `generic_selector( alpha^gen1 * (1 * w[0])) = w[0]`

Reply:
indeed the tests pass now, so that must be that I'm right :P

Reply:
Ah, indeed. My mistake!